q2_fragment_insertion/_insertion.py

```diff
@@ -302,3 +303,43 @@ def classify_otus_experimental(
              "i.e. are results from the same 'sepp' run."))
 
     return pd_taxonomy.set_index('Feature ID')
+
+
+def filter_features(table: biom.Table,
+                    tree: NewickFormat) -> (biom.Table, biom.Table):
+
+    # load the insertion tree
+    tree = skbio.TreeNode.read(str(tree))
+    # collect all tips=inserted fragments+reference taxa names
+    fragments_tree = {
+        str(tip.name)
+        for tip in tree.tips()
+        if tip.name is not None}
+
+    # collect all fragments/features from table
+    fragments_table = set(map(str, table.ids(axis='observation')))
```

Comment:
Why cast to `str`? It should be fine to just do `set(table.ids(axis='observation'))`

Reply:
I am afraid of people with feature-tables that only contain numbers, which might be parsed as `int` by the biom library, but the same numbers might be `str` in the phylogeny. Note that the tip names are the headers of the fasta sequences of the rep-set.qza artefact, thus it is not guaranteed that they have to be (nucelotide) strings.